tools/api-markdown-documenter/src/renderers/markdown-renderer/test/RenderHeading.test.ts

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { expect } from "chai";
+
+import { HeadingNode } from "../../../documentation-domain/index.js";
+import { testRender } from "./Utilities.js";
+
+describe("Heading Markdown rendering tests", () => {
+	describe("Standard context", () => {
+		it("Without ID", () => {
+			const input = HeadingNode.createFromPlainText("Hello World");
+			const result = testRender(input);
+			const expected = ["", "# Hello World", "", ""].join("\n");
+			expect(result).to.equal(expected);
+		});
+
+		it("With ID", () => {
+			const input = HeadingNode.createFromPlainText("Hello World", /* id: */ "heading-id");
+			const result = testRender(input);
+			const expected = ["", "# Hello World {#heading-id}", "", ""].join("\n");
+			expect(result).to.equal(expected);
+		});
+
+		it("With ID - includes content that must be escaped for Markdown", () => {
+			const input = HeadingNode.createFromPlainText("Hello World", /* id: */ "_heading-id_");
+			const result = testRender(input);
+			const expected = ["", "# Hello World {#\\_heading-id\\_}", "", ""].join("\n");
+			expect(result).to.equal(expected);
+		});
+	});
+
+	describe("Table context", () => {
```

Comment:
Is it worth having a case for the escaped content here too?

Reply:
I suppose it wouldn't hurt, but the scenario won't cause issues in HTML since the ID ends up being quoted regardless.

Reply:
And then I noticed that none of the table context tests were written to correctly render in a table context... fixing those tests now.